generative_ai/function_calling/advanced_example.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from vertexai.generative_models import GenerationResponse
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_function_call_advanced() -> GenerationResponse:
+    # [START generativeaionvertexai_gemini_function_calling_advanced]
+    import vertexai
+    from vertexai.preview.generative_models import (
+        FunctionDeclaration,
+        GenerativeModel,
+        Tool,
+        ToolConfig,
+    )
+
+    # TODO(developer): Update & uncomment below line
+    # PROJECT_ID = "your-project-id"
+
+    # Initialize Vertex AI
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    # Specify a function declaration and parameters for an API request
+    get_product_sku_func = FunctionDeclaration(
+        name="get_product_sku",
+        description="Get the available inventory for a Google products, e.g: Pixel phones, Pixel Watches, Google Home etc",
+        # Function parameters are specified in OpenAPI JSON schema format
```

Comment:
I was discussing this with some folks on the OpenAPI schema technical steering committee, and they suggested we refer to this as `JSON schema format` and drop the OpenAPI part for technical accuracy. So I suggest making that change here and throughout all Function Calling code samples. And I'm working on getting this updated in the Function Calling documentation as well.

Reply:
Done